source/speechViewer.py

```diff
@@ -93,10 +100,11 @@ def appendText(text):
 	_guiFrame.textCtrl.AppendText(text + "\n")
 
 def _cleanup():
-	global _guiFrame, isActive
+	global _guiFrame, _onActiveChanged, isActive
 	if not isActive:
 		return
 	isActive=False
+	_onActiveChanged(isActive)

```

Comment:
I understand the desire to use dependency injection here, rather than calling a global. However, I'm concerned that some external code might call `activate` and not pass a callable, thus breaking the invariant that the check box should be checked while the Speech Viewer is open and unchecked when it's closed. There are ways we can redesign this to be a bit cleaner, but for now, I think calling `gui.mainFrame.onSpeechViewerEnabled` is clearer.